lib/working_hours/computation.rb

```diff
@@ -183,20 +182,25 @@ def working_time_between from, to, config: nil
         to = in_config_zone(to, config: config)
         distance = 0
         while from < to
+          from_was = from
           # look at working ranges
           time_in_day = from.seconds_since_midnight
           config[:working_hours][from.wday].each do |begins, ends|
             if time_in_day >= begins and time_in_day < ends
-              # take all we can
-              take = [ends - time_in_day, to - from].min
-              # advance time
-              from += take
-              # increase counter
-              distance += take
+              if (to - from) > (ends - time_in_day)
+                # take all the range and continue
+                distance += (ends - time_in_day)
+                from = from.beginning_of_day + ends
+              else
+                # take only what's needed and stop
+                distance += (to - from)
+                from = to
+              end
             end
           end
           # roll to next business period
           from = advance_to_working_time(from, config: config)
+          raise "Invalid loop detected in working_time_between (from=#{from.iso8601(12)}, to=#{to.iso8601(12)}, distance=#{distance}, config=#{config}), please open an issue ;)" unless from > from_was
```

Comment:
Small addition to try to detect infinite loop here (I couldn't manage to reproduce one but I think with a good combination of times with nanoseconds it could have been possible). It detects when the loop does not advance in time basically and stop instead of looping forever, logging the parameters so we can reproduce.